# HG changeset patch
# User James Le Cuirot <chewi@aura-online.co.uk>
# Date 1276623161 -3600
# Node ID 1feb7c10b9ae5669721fad0aec3c7fb9f4ccfd78
# Parent  3806789df5b04b632bcc3a483f4668edcb164fbc
Use /usr/share/games/<appName> for data path on Gentoo.

diff -r 3806789df5b0 -r 1feb7c10b9ae BBGE/Core.cpp
--- a/BBGE/Core.cpp	Tue Jun 15 18:21:12 2010 +0100
+++ b/BBGE/Core.cpp	Tue Jun 15 18:32:41 2010 +0100
@@ -1008,25 +1008,10 @@
 		else
 			debugLog("Failed to chdir to filesystem path" + filesystem);
 	}
-	char path[PATH_MAX];
-	// always a symlink to this process's binary, on modern Linux systems.
-	const ssize_t rc = readlink("/proc/self/exe", path, sizeof (path));
-	if ( (rc == -1) || (rc >= sizeof (path)) )
-	{
-		// error!
-		debugLog("readlink");
-	}
-	else
-	{
-		path[rc] = '\0';
-		char *ptr = strrchr(path, '/');
-		if (ptr != NULL)
-		{
-			*ptr = '\0';
-			debugLog(path);
-			chdir(path);
-		}
-	}
+	std::string path("/usr/share/games/");
+	path.append(appName);
+	debugLog(path);
+	chdir(path.c_str());
 #endif
 #ifdef BBGE_BUILD_WINDOWS
 	// FIXME: filesystem not handled
